File: Chapter 05/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter05/section_5_6_crypto_tracker/after/infrastructure/adapters/FakePriceProvider.java
package com.grokkingsoftwarearchitecture.chapter05.section_5_6_crypto_tracker.after.infrastructure.adapters;

import com.grokkingsoftwarearchitecture.chapter05.section_5_6_crypto_tracker.after.core.ports.PriceProviderPort;
/**
 * ADAPTER 1: The "Airplane Mode" / Test Adapter.
 */
public class FakePriceProvider implements PriceProviderPort {
    private final double fixedPrice;

    public FakePriceProvider(double fixedPrice) {
        this.fixedPrice = fixedPrice;
    }

    @Override
    public double getBitcoinPrice() {
        return fixedPrice;
    }
}

File: Chapter 05/C#/Section_5_6_Crypto_Tracker/After/Infrastructure/Adapters/FakePriceProvider.cs
using System.Threading.Tasks;
using Chapter05.CryptoTracker.After.Core.Ports;

namespace Chapter05.CryptoTracker.After.Infrastructure.Adapters
{
    /// <summary>
    /// ADAPTER 1: The "Airplane Mode" / Test Adapter.
    /// Proves that we can run and test the system without an internet connection.
    /// </summary>
    public class FakePriceProvider : IPriceProviderPort
    {
        private readonly decimal _fixedPrice;

        public FakePriceProvider(decimal fixedPrice = 50_000m)
        {
            _fixedPrice = fixedPrice;
        }

        public Task<decimal> GetBitcoinPrice() => Task.FromResult(_fixedPrice);
    }
}